docs/execution-providers/CoreML-ExecutionProvider.md

```diff
@@ -49,18 +60,82 @@ Ort::ThrowOnError(OrtSessionOptionsAppendExecutionProvider_CoreML(so, coreml_fla
+```c++
+std::unordered_map<std::string, std::string> provider_options;
+provider_options["ModelFormat"] = std::to_string("MLProgram");
+provider_options["MLComputeUnits"] = std::to_string("ALL");
+provider_options["RequireStaticInputShapes"] = std::to_string("0");
+provider_options["EnableOnSubgraphs"] = std::to_string("0");
+```
+
+Python inference example code to use the CoreML EP run time options:
+```python
+import onnxruntime as ort
+model_path = "model.onnx"
+providers = [
+    ('CoreMLExecutionProvider', {
+        "ModelFormat": "MLProgram", "MLComputeUnits": "ALL", 
+        "RequireStaticInputShapes": "0", "EnableOnSubgraphs": "0"
+    }),
+]
+
+session = ort.InferenceSession(model_path, providers=providers)
+outputs = ort_sess.run(None, input_feed)
+```
+
+### Available Options (NEW API)
+`ModelFormat` can be one of the following values: (`NeuralNetwork` by default )
+- `MLProgram`: Create an MLProgram format model. Requires Core ML 5 or later (iOS 15+ or macOS 12+).
+- `NeuralNetwork`: Create a NeuralNetwork format model. Requires Core ML 3 or later (iOS 13+ or macOS 10.15+).
+
+`MLComputeUnits` can be one of the following values: (`ALL` by default )
+- `CPUOnly`: Limit CoreML to running on CPU only.
+- `CPUAndNeuralEngine`: Enable CoreML EP for Apple devices with a compatible Apple Neural Engine (ANE).
+- `CPUAndGPU`: Enable CoreML EP for Apple devices with a compatible GPU.
+- `ALL`: Enable CoreML EP for all compatible Apple devices.
+
+`RequireStaticInputShapes` can be one of the following values: (`0` by default )
+
+Only allow the CoreML EP to take nodes with inputs that have static shapes.
+By default the CoreML EP will also allow inputs with dynamic shapes, however performance may be negatively impacted by inputs with dynamic shapes.
+
+- `0`: Allow the CoreML EP to take nodes with inputs that have dynamic shapes.
+- `1`: Only allow the CoreML EP to take nodes with inputs that have static shapes.
+
+
+`EnableOnSubgraphs` can be one of the following values: (`0` by default )
+
+Enable CoreML EP to run on a subgraph in the body of a control flow operator (i.e. a [Loop](https://github.com/onnx/onnx/blob/master/docs/Operators.md#loop), [Scan](https://github.com/onnx/onnx/blob/master/docs/Operators.md#scan) or [If](https://github.com/onnx/onnx/blob/master/docs/Operators.md#if) operator).
+- `0`: Disable CoreML EP to run on a subgraph in the body of a control flow operator.
+- `1`: Enable CoreML EP to run on a subgraph in the body of a control flow operator.
+
+`SpecializationStrategy`:  This feature is available since MacOs>=10.15 or IOS>=18.0. This process can affect the model loading time and the prediction latency. Use this option to tailor the specialization strategy for your model. Navigate to [Apple Doc](https://developer.apple.com/documentation/coreml/mloptimizationhints-swift.struct/specializationstrategy-swift.property) for more information. Can be one of the following values: (`Default` by default )
```

Comment:
```suggestion
`SpecializationStrategy`:  This feature is available since macOS>=10.15 or iOS>=18.0. This process can affect the model loading time and the prediction latency. Use this option to tailor the specialization strategy for your model. Navigate to [Apple Doc](https://developer.apple.com/documentation/coreml/mloptimizationhints-swift.struct/specializationstrategy-swift.property) for more information. Can be one of the following values: (`Default` by default )
```